### ISSUE TITLE
plus/minus functions don't seem to support fractional inputs

### ISSUE THREAD

**Issue by leebickmtu** [moment/luxon#409]
`DateTime.local().plus({days: 0.8})`

Fractional values for a unit, like above, don't seem to work. My use case is that I'm getting a value that is always normalized to days from an API and want to add it to the current time.

I thought maybe I could get around this by using `Duration.fromObject({days: 0.8}).normalize()`, but `normalize` also doesn't convert it to the whole number hour/minute units either.

**Comment by icambron** [comment]
Yeah, I think `plus()` ought to work here, adding 3/5 of a day.

`normalize` isn't quite as obvious, but at first blush I think you're right that it should do the same.

As a workaround, you can use `Duration.fromObject({days: 0.8}).shiftTo("hours")`

**Comment by icambron** [comment]
Note to my future self: Probably best would be to normalize down the fractions so that all the highest-order parts are integers, and then convert the lowest calendar unit's remainder to millis, in order to maximize the amount of calendar math we can do with the downshifted units.

**Comment by JaredReisinger** [comment]
For what it's worth, I've run into the same underlying problem in parsing user input into a `Duration`... `Duration.fromObject({ hours: 2.5 })` results in a "hard-to-use" object.  My workaround was to use `Duration.shiftTo()` to clean things up.  I would have simply used `Duration.normalize()`, but it attempts to keep the same units as the original object, and won't (for example) create the "minutes" property to take the fractional hours.  (It also only seems to correct for overflow; `Duration.fromObject({ hours: 2.5, minutes 1 }).normalize()` does not break apart the fractional hour.)

But in any case, `Duration.fromObject({days: 0.8}).shiftTo('days', 'hours', 'minutes', 'seconds', 'milliseconds')` results in a Duration with `{ days: 0, hours: 19, minutes: 12, seconds: 0, milliseconds:  1.0231815394945443e-8 }`, and _that_ works for `DateTime.plus()` nicely.

I think that `Duration.shiftTo('years', 'days', 'hours', 'minutes', 'seconds', 'milliseconds')` would "canonicalize" a Duration pretty well.

**Comment by icambron** [comment]
You can't quite do that. For example

```js
> Duration.fromObject({ hours: 24 }).shiftTo("days", "hours").toObject() // => { days: 1, hours: 0 }
```

which would result in very different math across DSTs. Similar problems with days -> years because of leap years. Throw months into the mix and it's a mess.

I think the decimals need to be considered explicitly.

**Comment by icambron** [comment]
Fixed in ee70b8b and released in 1.18.2

**Comment by leebickmtu** [comment]
Just tested, looks good.
